app/src/main/java/org/mozilla/focus/exceptions/ExceptionMigrationMiddleware.kt

```diff
@@ -35,8 +37,10 @@ class ExceptionMigrationMiddleware(
     ) {
         next(action)
 
-        if (action is TabListAction) {
+        if (action is TabListAction || action is ToggleExclusionListAction) {
             checkIfMigrationIsNeeded(context.state)
+        } else if (action is InitAction) {
```

Comment:
This was added so that the list would be synced if you go directly to settings without opening a tab.

Reply:
Ah, I see. I didn't add it initially to avoid migrating twice, what we would do now if you open a tab?

Reply:
Maybe we can remove `action is TabListAction` from the first if, so we would not migrate again. We would need this migration just at app start and when toggling etp for a session (ToggleExclusionListAction)

Reply:
We also need it when all `GeckoSession` instances are closed. Once all "private" ones are gone, the private context is gone and the migrated exceptions are lost. That's why we do it before the opening of a new tab.